### ISSUE TITLE
Feature Request: `profile` output should be sorted

### ISSUE THREAD

**Issue by steven-johnson** [halide/Halide#7638]
It would be a bit easier to read the `profile` output if `halide_profiler_report_unlocked()` sorted the output by percentage.

**Comment by abadams** [comment]
The downside of that is it would make the ordering unstable from run to run, making comparisons difficult. Currently it's in rough order of realization, which is also annoying, because it's sensitive to the schedule.

Possibly it should be in pipeline order, so early stages are always at the top, and later stages are always at the bottom.

**Comment by steven-johnson** [comment]
> would make the ordering unstable from run to run

True, but usually when I'm profiling I want to start by looking at the biggest costs; this makes them easier to spot. (Maybe consider adding it via an env var?)

> in pipeline order

Is that information transmitted to the profiler right now? if not, how hard would it be?
